### PR TITLE
Use HttpClientFactory

### PR DESCRIPTION
**Pull request by benaadams**
Use `HttpClientFactory` when `MessageHandler` not specified

/cc @glennc

### CODE REVIEW

File: src/Microsoft.AspNetCore.Proxy/ProxyService.cs [hunk 1]
```diff
@@ -9,18 +9,28 @@ namespace Microsoft.AspNetCore.Proxy
 {
     public class ProxyService
     {
-        public ProxyService(IOptions<SharedProxyOptions> options)
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly HttpClient _httpClient;
+
+        public ProxyService(IOptions<SharedProxyOptions> options, IHttpClientFactory httpClientFactory)
```
[the remark this answers is not in the document]
Comment (jkotalik) [reply src/Microsoft.AspNetCore.Proxy/ProxyService.cs]: I wasn't sure myself either, so I ran this locally. Seems that the larger constructor with the IHttpClientFactory is still called after adding the other constructor.

File: src/Microsoft.AspNetCore.Proxy/ProxyService.cs [hunk 2]
```diff
@@ -9,18 +9,28 @@ namespace Microsoft.AspNetCore.Proxy
...
         {
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
             Options = options.Value;
-            Client = new HttpClient(Options.MessageHandler ?? new HttpClientHandler { AllowAutoRedirect = false, UseCookies = false });
```
Comment (jkotalik) [inline src/Microsoft.AspNetCore.Proxy/ProxyService.cs]: @Tratcher how important is it for these options to be preserved. The HttpClientFactory doesn't have a great way of specifying HttpClientHandler options. See https://github.com/aspnet/HttpClientFactory/issues/110

Comment (mkosieradzki) [reply src/Microsoft.AspNetCore.Proxy/ProxyService.cs]: There options are critical. First one means the proxy want be able to properly forward 301 and 302 responses. Lack of the Second one hurts when using the legacy handler and ruins cookie forwarding to the server.

**Comment by Tratcher** [comment]
We're not currently taking PRs in this repo, unless @rynowak and @glennc wanted this as a proof of concept?

**Comment by natemcmaster** [comment]
I'm closing this PR as we are archiving this repo and moving all prototypes like this one into https://github.com/aspnet/asplabs. If you are still interested in contributing this change, please re-open your PR against https://github.com/aspnet/asplabs.

Thanks,
Nate

**Pull request closed without merging**